### DISCUSSION TITLE
(Solved) How to loop through locals and list at the same time to generate resources

### DISCUSSION THREAD

**Discussion by JavierLopezT** [Snowflake-Labs/terraform-provider-snowflake#299] [Q&A]
EDIT: I solved the problem but I am not able to close the discussion

I have the following tf file:

```
locals {
      schemas = {
        "ODS" = {
          usage_roles = ["TRANSFORMER"]
        }
        "EXT" = {
          usage_roles = []
        }
        "INT" = {
          usage_roles = ["REPORTER"]
        }
        "DW" = {
          usage_roles = ["LOADER"]
        }
      }
    }
    
    resource "snowflake_schema" "schema" {
      for_each = local.schemas
      name = each.key
      database = ???????
      usage_roles = each.value.usage_roles
    }
```

I want to maintain the locals as it is (different usage_roles for each schema and hardcoded here) while having several values as database for each schema. In pseudo-code it would be:
```
    for database in ['db_1', 'db_2', 'db_3']:
        resource "snowflake_schema" "schema" {
          for_each = local.schemas
          name = each.key
          database = database
          usage_roles = each.value.usage_roles
        }
```
So that we have the same schema resource in the three different databases. I have read some articles that point me to the belief that it is possible to make this loop but pre assigning all the values, meaning that I would have to put usage_roles in a list or something instead of hardcoding here in locals, which I think is less readable. For instance: 
https://stackoverflow.com/questions/57570505/terraform-how-to-use-for-each-loop-on-a-list-of-objects-to-create-resources

Is even possible what I am asking for? If so, how? Thank you very much in advance

**Comment by Mouftizo** [comment]
Have you managed to find a solution this? I am looking for the same thing : )

**Reply by JavierLopezT** [reply]
Hi, yes! You can see the answer here https://stackoverflow.com/questions/64756607/how-to-loop-through-locals-and-list-at-the-same-time-to-generate-resources
